File: Napakalaki/src/napakalaki/Napakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author carlos
 */
public class Napakalaki {
    private static final Napakalaki instance = new Napakalaki();
    private CombatResult result;
    
    //Variables de otras clases
    private Monster currentMonster;
    private Player currentPlayer;
    private ArrayList <Player> players = new ArrayList();
    
    private Napakalaki(){
        
    }
    
    private void initPlayers(ArrayList<String> names){
    
    }
    
    private Player nextPlayer(){
        
        return null;
    }
    
    private boolean nextTurnAllowed(){
    
        return false;
    }
    
    private void setEnemies(){
    
    }
    
    public static Napakalaki getInstance(){
        
        return null;
    }
    
    public CombatResult developCombat(){
        return null;
    }
    
    public void discardVisibleTreasures(ArrayList<Treasure>treasures){
    
    }
    
    public void discardHiddenTreasures(ArrayList<Treasure>treasures){
    
    }
    
    public void makeTreasureVisible(ArrayList<Treasure>treasure){
    
    }
    
    public void initGame(ArrayList<String>players){
    
    }
    

    public Player getCurrentPlayer(){
    
        return null;
    }
     public Monster getCurrentMonster(){
     
         return null;
    }
    
    public boolean nextTurn(){
    
        return false;
    } 
    
    public boolean endOfGame(CombatResult result){
    
        return false;
    }
}

File: Napakalaki-Ruby/lib/napakalaki.rb
# Javier Aranda Izquierdo
module NapakalakiGame
  
  require "singleton"

  class Napakalaki

    include Singleton

    def initialize(resultado,currentMonster,dealer,currentPlayer,players)
      @result=resultado
      @currentMonster=currentMonster
      @dealer=dealer
      @currentPlayer=currentPlayer
      @players=players

    end

      private def Napakalaki

      end

      private def initPlayers(names)

      end

      private def nextPlayer

      end

      private def nextTurnAllowed

      end

      private def setEnemies

      end

      def developCombat

      end

      def discardVisibleTreasures(treasures)

      end

      def discardHiddenTreasures(treasures)

      end

      def makeTreasuresVisible(treasures)

      end

      def initGame(players)

      end

      def getCurrentPlayer

      end

      def getCurrentMonster

      end

      def nextTurn

      end

      def endOfGame(result)

      end
  end
end